tests_pytest/common_tests/unit_tests/core/graph/test_base_graph.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2025 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from copy import deepcopy 
+
+import pytest
+from unittest.mock import Mock
+
+from mct_quantizers import QuantizationMethod
+from model_compression_toolkit.core.common import Graph
+from model_compression_toolkit.core.common.graph.base_node import BaseNode
+from model_compression_toolkit.core.common.fusion.fusing_info import FusingInfo
+from model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema import Signedness
+from tests.common_tests.helpers.generate_test_tpc import generate_test_attr_configs, generate_test_op_qc
+from model_compression_toolkit.core.common.quantization.node_quantization_config import ActivationQuantizationMode, NodeActivationQuantizationConfig
+from model_compression_toolkit.core.common.quantization.candidate_node_quantization_config import CandidateNodeQuantizationConfig
+from model_compression_toolkit.core.common.quantization.quantization_params_generation.power_of_two_selection import power_of_two_selection_histogram
+
+    
+# Setup TEST_QC and TEST_QCO for testing.
+TEST_QC = generate_test_op_qc(**generate_test_attr_configs(),
+                              activation_n_bits=16,
+                              activation_quantization_method=QuantizationMethod.POWER_OF_TWO)
+
+def build_mock_fusing_info(nodes):
+    """
+    Creates a mock FusingInfo object that simulates the behavior of fusing information in a graph.
+    """
+
+    fusing_info = Mock(spec=FusingInfo)
+    fusing_info.get_inner_fln_nodes.return_value = [nodes[0], nodes[1]]
```

Comment:
I suggest to split this into two test ssenarios - 
1. FLN quant wuth expected quantization
2. FLN with expected quantization disabled

also, please add a case where the OpQuantizationConfig is not None, but activation quantization is disabled in the config (should get "FLN_NO_QUANT" in the test)